packages/blockchain/src/db/dbManager.ts

```diff
@@ -129,84 +111,84 @@ export class DBManager {
-    const key = tdKey(number, hash)
-    const td = await this.get(key, { cache: 'td' })
+  async getTd(blockHash: Buffer, blockNumber: BN): Promise<BN> {
+    const td = await this.get(DatabaseOperationTarget.TotalDifficulty, { blockHash, blockNumber })
     return new BN(rlp.decode(td))
   }
 
   /**
    * Performs a block hash to block number lookup.
    */
-  async hashToNumber(hash: Buffer): Promise<BN> {
-    const key = hashToNumberKey(hash)
-    const value = await this.get(key, { cache: 'hashToNumber' })
+  async hashToNumber(blockHash: Buffer): Promise<BN> {
+    const value = await this.get(DatabaseOperationTarget.HashToNumber, { blockHash })
     return new BN(value)
   }
 
   /**
    * Performs a block number to block hash lookup.
    */
-  async numberToHash(number: BN): Promise<Buffer> {
-    if (number.ltn(0)) {
+  async numberToHash(blockNumber: BN): Promise<Buffer> {
+    if (blockNumber.ltn(0)) {
       throw new level.errors.NotFoundError()
     }
 
-    const key = numberToHashKey(number)
-    return this.get(key, { cache: 'numberToHash' })
+    return this.get(DatabaseOperationTarget.NumberToHash, { blockNumber })
   }
 
   /**
    * Fetches a key from the db. If `opts.cache` is specified
    * it first tries to load from cache, and on cache miss will
    * try to put the fetched item on cache afterwards.
    */
-  async get(key: string | Buffer, opts: GetOpts = {}): Promise<any> {
-    const dbOpts = {
-      keyEncoding: opts.keyEncoding || 'binary',
-      valueEncoding: opts.valueEncoding || 'binary',
-    }
+  async get(databaseOperationTarget: DatabaseOperationTarget, key?: DatabaseKey): Promise<any> {
+    const databaseGetOperation = DatabaseOperation.get(databaseOperationTarget, key)
 
-    if (opts.cache) {
-      if (!this._cache[opts.cache]) {
-        throw new Error(`Invalid cache: ${opts.cache}`)
+    const cacheString = databaseGetOperation.cacheString
+    const dbKey = databaseGetOperation.baseDBOp.key
+    const dbOpts = databaseGetOperation.baseDBOp
+
+    if (cacheString) {
+      if (!this._cache[cacheString]) {
+        throw new Error(`Invalid cache: ${cacheString}`)
       }
 
-      let value = this._cache[opts.cache].get(key)
+      let value = this._cache[cacheString].get(dbKey)
       if (!value) {
-        value = <Buffer>await this._db.get(key, dbOpts)
-        this._cache[opts.cache].set(key, value)
+        value = <Buffer>await this._db.get(dbKey, dbOpts)
+        this._cache[cacheString].set(dbKey, value)
       }
 
       return value
     }
 
-    return this._db.get(key, dbOpts)
+    return this._db.get(dbKey, dbOpts)
   }
 
   /**
    * Performs a batch operation on db.
    */
-  async batch(ops: DBOp[]) {
-    return this._db.batch(ops as any)
+  async batch(ops: DatabaseOperation[]) {
+    const convertedOps: DBOp[] = ops.map((op) => op.baseDBOp)
+    // update the current cache for each operation
+    ops.map((op) => op.updateCache(this._cache))
```

Comment:
Is this a substantive change (a bug fix)? (Am just requesting more info - I don't know this lib very well)

Reply:
This PR does not fix any bugs. Previously, these cache operations were all handled in `index.ts` of blockchain, but now they are handled within these `DatabaseOperation`s every time we do a batch request of these `ops`.